packages/core/src/application_init.ts

```diff
@@ -72,7 +72,7 @@ import {noop} from './util/noop';
  *    bootstrap: [AppComponent],
  *    providers: [{
  *      provide: APP_INITIALIZER,
- *      useFactory: initializeApp,
+ *      useFactory: (httpClient: HttpClient) => initializeApp(httpClient),
```

Comment:
Actually...looking at this more carefully, I think my [previous suggestion][1] wasn't a good one 😅
I thought that we could get rid of the changes to `initializeApp()`, but I see now that we still need for it to return a function (and not the observable directly).

Given that, I think it is better to revert this change:
```suggestion
 *      useFactory: initializeApp,
```

Sorry for the inconvenience 😇

##
In addition to reverting this change, maybe we could rename the function to `initializeAppFactory` to make it more clear that this is a factory function and not the initializer (in contrast to the previous example).

WDYT, @skyfranchittog?

[1]: https://github.com/angular/angular/pull/42743#discussion_r662908881

Reply:
I agree with you @gkalpak, it is better to make this setup as clear as possible ;)